pkg/infra/game/agent/agent.go

```diff
@@ -15,6 +15,19 @@ type Agent struct {
 	Strategy
 }
 
+func (a *Agent) HandleDonateToHpPool(agentState state.AgentState) uint {
+	// fmt.Print(agentState)
+	a.BaseAgent.latestState = agentState
+
+	return a.Strategy.DonateToHpPool(a.BaseAgent)
+}
+
+func (a *Agent) HandleUpdateInternalState(agentState state.AgentState, fightResults *commons.ImmutableList[decision.ImmutableFightResult], voteResults *immutable.Map[decision.Intent, uint]) {
```

Comment:
Should this include HPPool donation results?

Reply:
Maybe not, maybe the hp pool value should just be in view? idk if agents are meant to see how much others donate?